src/data/markdown/docs/30 xk6-browser/01 xk6-browser/02-browser-context.md

```diff
@@ -5,27 +5,69 @@ excerpt: "xk6-browser: BrowserContext Class"
-| <a href="https://playwright.dev/docs/api/class-browsercontext#browser-context-clear-permissions" target="_blank">browserContext.clearPermissions()</a> |   |
-| <a href="https://playwright.dev/docs/api/class-browsercontext#browser-context-grant-permissions" target="_blank">browserContext.grantPermissions(permissions[, options])</a> |   |
-| <a href="https://playwright.dev/docs/api/class-browsercontext#browser-context-close" target="_blank">browserContext.close()</a> |   |
-| <a href="https://playwright.dev/docs/api/class-browsercontext#browser-context-new-page" target="_blank">browserContext.newPage()</a> |   |
-| <a href="https://playwright.dev/docs/api/class-browsercontext#browser-context-pages" target="_blank">browserContext.pages()</a> |   |
-| <a href="https://playwright.dev/docs/api/class-browsercontext#browser-context-set-default-navigation-timeout" target="_blank">browserContext.setDefaultNavigationTimeout(timeout)</a> |   |
-| <a href="https://playwright.dev/docs/api/class-browsercontext#browser-context-set-default-timeout" target="_blank">browserContext.setDefaultTimeout(timeout)</a> |   |
-| <a href="https://playwright.dev/docs/api/class-browsercontext#browser-context-set-geolocation" target="_blank">browserContext.setGeolocation(geolocation)</a> |   |
-| <a href="https://playwright.dev/docs/api/class-browsercontext#browser-context-set-http-credentials" target="_blank">browserContext.setHTTPCredentials(httpCredentials)</a> |   |
-| <a href="https://playwright.dev/docs/api/class-browsercontext#browser-context-set-offline" target="_blank">browserContext.setOffline(offline)</a> |   |
-
-### Missing Playwright APIs
+```javascript
+import launcher from 'k6/x/browser';
+
+export default function () {
+  const browser = launcher.launch('chromium', { headless: false });
+  const context = browser.newContext();
+  const page = context.newPage();
+  page.goto('http://whatsmyuseragent.org/');
+  page.close();
+  browser.close();
+}
+```
+
+
+
+- [browserContext.addInitScript(script[, arg])](#browsercontext-addinitscript-script-arg)
+- [browserContext.clearCookies()](#browsercontext-clearcookies)
+- [browserContext.clearPermissions()](#browsercontext-clearpermissions)
+- [browserContext.grantPermissions(permissions[, options])](#browsercontext-grantpermissions-permissions-options)
+- [browserContext.close()](#browsercontext-close)
+- [browserContext.newPage()](#browsercontext-newpage)
+- [browserContext.pages()](#browsercontext-pages)
+- [browserContext.setDefaultNavigationTimeout(timeout)](#browsercontext-setdefaultnavigationtimeout-timeout)
+- [browserContext.setDefaultTimeout(timeout)](#browsercontext-setdefaulttimeout-timeout)
+- [browserContext.setGeolocation(geolocation)](#browsercontext-setgeolocation-geolocation)
+- [browserContext.setHTTPCredentials(httpCredentials)](#browsercontext-sethttpcredentials-httpcredentials)
```

Comment:
This method now comes with a deprecation warning in v0.2.0. I wanted to let you know, see: https://github.com/grafana/xk6-browser/pull/242/commits/76e7ea273d13a311c699b2eb82a399a8143713ab.